### PR TITLE
add console command for account creation

### PR DESCRIPTION
**Pull request by ChristophWurst**
This adds a simple command as described in #1125.

implements #1125 

TODO:
- [x] encrypt passwords
- [x] add unit test

@DeepDiver1975 @hitam4450 @owncloud/mail

### CODE REVIEW

**Comment by ChristophWurst** [comment]
Except for the encryption of passwords, this is already usable like this:
```bash
php occ mail:account:create user 'Example User' user@example.com imap.example.com 993 ssl user@example.com 123456 smtp.example.com 465 ssl user@example.com 123456
```

**Comment by ChristophWurst** [comment]
@DeepDiver1975 I just found your commit 4b35698. Please open a WIP-branch the next time so we don't implement stuff twice ;-)

**Comment by ChristophWurst** [comment]
Travis reported an error with core stable7:
```
php occ app:enable mail

An unhandled exception has been thrown:

exception 'InvalidArgumentException' with message 'Identifier "OCA\Mail\Service\AccountService" is not defined.' in /home/travis/build/owncloud/core/3rdparty/Pimple/Pimple.php:78

Stack trace:

#0 /home/travis/build/owncloud/core/lib/private/appframework/utility/simplecontainer.php(17): Pimple->offsetGet('OCA\\Mail\\Servic...')

#1 /home/travis/build/owncloud/core/apps/mail/appinfo/register_command.php(16): OC\AppFramework\Utility\SimpleContainer->query('OCA\\Mail\\Servic...')

#2 /home/travis/build/owncloud/core/console.php(42): require('/home/travis/bu...')

#3 /home/travis/build/owncloud/core/occ(11): require_once('/home/travis/bu...')
```

File: appinfo/register_command.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+ * later. See the COPYING file.
+ *
+ * @author Christoph Wurst <christoph@winzerhof-wurst.at>
+ * @copyright Christoph Wurst 2016
+ */
+
+use OC;
+use OCA\Mail\Command\CreateAccount;
+
+$accountService = OC::$server->query('OCA\Mail\Service\AccountService');
```
Comment (DeepDiver1975) [inline appinfo/register_command.php]: the account service is not registered in the central server container - you need to get the mail application container for this .... to make it work for oc7

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 I just found your commit 4b35698. Please open a WIP-branch the next time so we don't implement stuff twice ;-)

not really worth - 5 minutes job :wink:

File: appinfo/register_command.php (hunk 1 above)
Comment (ChristophWurst) [inline appinfo/register_command.php]: I'm aware of that and so I'm planning to leave this for 0.4 where oc7 support is dropped.

File: lib/command/createaccount.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+		$account->setOutboundHost($smtpHost);
+		$account->setOutboundPort($smtpPort);
+		$account->setOutboundSslMode($smtpSslMode);
+		$account->setOutboundUser($smtpUser);
+		$account->setOutboundPassword($smtpPassword);
+
+		$this->accountService->save($account);
+
+		$output->writeln("<info>Account $email created");
```
Comment (ChristophWurst) [inline lib/command/createaccount.php]: </info> is missing

File: appinfo/register_command.php (hunk 1 above)
Comment (DeepDiver1975) [inline appinfo/register_command.php]: :+1:

File: appinfo/register_command.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * ownCloud - Mail
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Christoph Wurst <christoph@winzerhof-wurst.at>
+ * @copyright Christoph Wurst 2016
+ */
+
+use OC;
```
Comment (DeepDiver1975) [inline appinfo/register_command.php]:
````
Warning: The use statement with non-compound name 'OC' has no effect in ./appinfo/register_command.php on line 13

**Comment by ChristophWurst** [comment]
@DeepDiver1975 another review please.

I did not write a test for the ``execute`` method as it's protected

**Comment by ChristophWurst** [comment]
I added a version-hack like in #1247

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**